Calculator/src/main/java/com/microsoft/device/display/samples/calculator/models/Calculations.kt

```diff
@@ -0,0 +1,133 @@
+package com.microsoft.device.display.samples.calculator.models
+
+import com.microsoft.device.display.samples.calculator.ui.pages.calculatorModel
+import kotlin.math.log10
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+fun add() {
+    calculatorModel.x = (calculatorModel.x.toFloat() + calculatorModel.y.toFloat()).toString()
+}
+
+fun sub() {
+    calculatorModel.x = (calculatorModel.x.toFloat() - calculatorModel.y.toFloat()).toString()
+}
+
+fun mul() {
+    calculatorModel.x = (calculatorModel.x.toFloat() * calculatorModel.y.toFloat()).toString()
+}
+
+fun div() {
+    calculatorModel.x = if (calculatorModel.y == "0") "0.0" else (calculatorModel.x.toFloat() / calculatorModel.y.toFloat()).toString()
+}
+
+fun pi() {
+    calculatorModel.overWrite = true
+    calculatorModel.addValues(Math.PI.toFloat().toString())
+}
+
+fun sin() {
+    if (calculatorModel.isOnX) {
+        calculatorModel.x = if (calculatorModel.degrees) kotlin.math.sin(
+            Math.toRadians(
+                calculatorModel.x.toDouble()
+            )
+        ).toFloat().toString() else kotlin.math.sin(calculatorModel.x.toDouble()).toFloat().toString()
+    } else {
+        calculatorModel.y = if (calculatorModel.degrees) kotlin.math.sin(
+            Math.toRadians(
+                calculatorModel.y.toDouble()
+            )
+        ).toFloat().toString() else kotlin.math.sin(calculatorModel.y.toDouble()).toFloat().toString()
+    }
+}
```

Comment:
optional style suggestion to improve readability/remove duplicate code (if you like this suggestion, you can probably do something similar for the other calculations that require similar conversions):
```suggestion
fun sin() {
    if (calculatorModel.isOnX) {
        calculatorModel.x = executeTrigFunction(::sin, calculatorModel.x)
    } else {
        calculatorModel.y = executeTrigFunction(::sin, calculatorModel.y)
    }
}

private fun executeTrigFunction(trigFunction: (Double) -> Double, rawInput: String): String {
    val radiansInput = if (calculatorModel.degrees) Math.toRadians(rawInput.toDouble()) else rawInput.toDouble()

    return trigFunction(radiansInput).toFloat().toString()
}
```

Note: for the suggestion above, had to add `import kotlin.math.sin`

Reply:
Thank you!! I was trying to figure out how to do something like this but couldn't figure it out.  Appreciate it